aiohttp_wsgi/wsgi.py

```diff
@@ -256,37 +262,40 @@ def _get_environ(self, request, body, content_length):
         # All done!
         return environ
 
-    async def handle_request(self, request):
+    @asyncio.coroutine
+    def handle_request(self, request):
         # Check for body size overflow.
         if request.content_length is not None and request.content_length > self._max_request_body_size:
             raise HTTPRequestEntityTooLarge()
         # Buffer the body.
-        async with ReadBuffer(self._inbuf_overflow, self._max_request_body_size, self._loop, self._executor) as body:
-            while True:
-                block = await request.content.readany()
-                if not block:
-                    break
-                await body.write(block)
-            # Seek the body.
-            body, content_length = await body.get_body()
-            # Get the environ.
-            environ = self._get_environ(request, body, content_length)
-            status, reason, headers, body = await self._loop.run_in_executor(
-                self._executor,
-                _run_application,
-                self._application,
-                environ,
-            )
-            # All done!
-            return Response(
-                status=status,
-                reason=reason,
-                headers=headers,
-                body=body,
-            )
-
-    async def __call__(self, request):
-        return await self.handle_request(request)
+        body = ReadBuffer(self._inbuf_overflow, self._max_request_body_size, self._loop, self._executor)
```

Comment:
If we're dropping the `async with`, then don't we need to use a try/catch to close the body when we're done?